client/src/components/MovieCard/MovieCard.js

```diff
@@ -1,12 +1,35 @@
+import { useEffect, useState } from 'react'
+import axios from 'axios'
 import styled from 'styled-components/macro'
 
-const MovieCard = ({ imgUrl, title, year, genre }) => {
+const MovieCard = ({ tmdbId }) => {
+  const [movie, setMovie] = useState({
+    tmdbId: '550',
+    imgUrl: 'https://image.tmdb.org/t/p/w500/a26cQPRhJPX6GbWfQbvZdrrp9j9.jpg',
+    title: 'Fight Club',
+    year: '1999',
+    genre: 'Drama',
+  })
+
+  useEffect(() => {
+    const fetchMovie = async id => {
+      try {
+        const response = await axios.get(`/api/tmdb/${id}`)
+        setMovie(response.data)
+      } catch (error) {
+        console.error(error)
+      }
+    }
+
+    fetchMovie(tmdbId)
```

Comment:
Probably you can give useEffect an async function directly, so you won't need the nested function anymore

Reply:
I googled it yesterday and I got a few articles on it why you shouldnt do it:
[https://www.robinwieruch.de/react-hooks-fetch-data](https://www.robinwieruch.de/react-hooks-fetch-data)
[https://dev.to/danialdezfouli/what-s-wrong-with-the-async-function-in-useeffect-4jne](https://dev.to/danialdezfouli/what-s-wrong-with-the-async-function-in-useeffect-4jne)